File: src/main/java/jukebot/commands/Stop.java
package jukebot.commands;

import jukebot.audioutilities.AudioHandler;
import jukebot.utils.Command;
import jukebot.utils.CommandProperties;
import jukebot.utils.Context;

@CommandProperties(description = "Ends the queue and current track", category = CommandProperties.category.CONTROLS)
public class Stop implements Command {

    public void execute(final Context context) {

        final AudioHandler player = context.getAudioPlayer();

        if (!player.isPlaying()) {
            context.sendEmbed("Not Playing", "Nothing is currently playing.");
            return;
        }

        if (!context.isDJ(true)) {
            context.sendEmbed("Not a DJ", "You need to be a DJ to use this command.\n[See here on how to become a DJ](https://jukebot.xyz/faq)");
            return;
        }

        player.getQueue().clear();
        player.playNext();

    }
}

File: src/main/java/jukebot/commands/Stop.kt
package jukebot.commands

import jukebot.audio.AudioHandler
import jukebot.framework.*

@CommandProperties(description = "Ends the queue and current track", category = CommandCategory.CONTROLS)
@CommandChecks.Dj(alone = true)
@CommandChecks.Playing
class Stop : Command(ExecutionType.REQUIRE_MUTUAL) {

    override fun execute(context: Context) {
        val player = context.getAudioPlayer()
        player.repeat = AudioHandler.RepeatMode.NONE
        player.queue.clear()
        player.playNext(false)
    }

}
